### PR TITLE
solved task

### PR DESCRIPTION
**Pull request by slaybrute**

### CODE REVIEW

**Comment by slaybrute** [comment]
It is a second time when I solve this task, hence it is changes after first verification

File: src/main/java/core/basesyntax/CreationOfFigure.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+public class CreationOfFigure extends FigureSupplier {
```
Comment (nacenik) [inline src/main/java/core/basesyntax/CreationOfFigure.java:3]: Class with method `main` shouldn't inherit any logic. FigureSupplier should be method-level field. Please, folow the rule [is-a, has-a](https://stackoverflow.com/questions/2218937/has-a-is-a-terminology-in-object-oriented-language)
```suggestion
public class Main {
```

File: src/main/java/core/basesyntax/CreationOfFigure.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    public static void main(String[] args) {
```
Comment (nacenik) [inline src/main/java/core/basesyntax/CreationOfFigure.java:4]: Create field FigureSupplier and Figure array with some size. Initialize every element in this field according to the task, and in another loop call method `draw` from every single initialized figure.

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                        random.nextInt(MAX_VALUE_FOR_FIGURES),
+                        random.nextInt(MAX_VALUE_FOR_FIGURES),
+                        random.nextInt(MAX_VALUE_FOR_FIGURES)),
+                new Square("square", getRandomColor(),
+                        random.nextInt(MAX_VALUE_FOR_FIGURES))};
+        return figures[random.nextInt(figures.length)];
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle("circle", "white", 10);
```
Comment (nacenik) [inline src/main/java/core/basesyntax/FigureSupplier.java:29]: [All magic number should be constant](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#all-magic-numbers-in-your-code-should-be-constants)

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+public class Figure implements PrintableInfo, SetableArea {
+    private String name;
```
Comment (nacenik) [inline src/main/java/core/basesyntax/Figure.java:4]: Create additional enum for this field **OR** delete this field and write the name of the figure inside the method `draw()`

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        this.firstBase = firstBase;
+        this.secondBase = secondBase;
+        this.hight = hight;
+        super.setArea((firstBase + secondBase) / 2.0);
+    }
+
+    @Override
+    public void printInfoAboutFigures() {
+        super.printInfoAboutFigures();
+        System.out.println("side: " + side + ", " + "first base: " + firstBase + ", "
```
[the remark this answers is not in the document]
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/IsoscelesTrapezoid.java:23]: not fixed

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 6]
```diff
@@ -0,0 +1,31 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    private static final int MAX_VALUE_FOR_FIGURES = 26;
+    private static final Random random = new Random();
+
+    public Figure getRandomFigure() {
```
[the remark this answers is not in the document]
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/FigureSupplier.java:10]: not fixed

File: src/main/java/core/basesyntax/Figure.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    private String color;
+
+    private double area;
```
[the remark this answers is not in the document]
Comment (Ivan95kos) [reply src/main/java/core/basesyntax/Figure.java:5]: not fixed

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 8]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class ColorSupplier {
+    private static final Random random = new Random();
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: Random does not have to be a constant.
```suggestion
    private Random random = new Random();
```

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    private static final int MAX_VALUE_FOR_FIGURES = 26;
+    private static final int MAX_VALUE_FOR_CIRCLE_RADIUS = 10;
+    private static final Random random = new Random();
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/FigureSupplier.java:8]: Random does not have to be a constant.
```suggestion
    private Random random = new Random();
```

**Comment by slaybrute** [comment]
I've fixed, I don't have any assumption why you've seen them. I'll commit again

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Please, fix previous comments
Especially this:
Знімок екрана 2023-01-17 о 20 45 25

File: src/main/java/core/basesyntax/Circle.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+        this.radius = radius;
+        super.setArea(Math.PI * Math.pow(radius, 2));
+    }
+
+    @Override
+    public void draw() {
+        System.out.print("Name: circle");
+        super.draw();
+        System.out.println("radius " + radius);
+    }
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Circle.java:17]: You should not call a method of the parent class.
It is necessary to override the implementation for each figure
Fix for each extends
```suggestion
    @Override
    public void draw() {
        System.out.println("description of the figure")
    }
```

File: src/main/java/core/basesyntax/Figure.java [hunk 11]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Figure implements Drawable, AreaCalculator {
+    private Color color;
+    private double area;
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Figure.java:5]: This variable is unnecessary. Delete it.
```suggestion
```

File: src/main/java/core/basesyntax/Figure.java [hunk 12]
```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Figure implements Drawable, AreaCalculator {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Figure.java:3]: This class should be abstract
```suggestion
public abstract class Figure implements Drawable, AreaCalculator {
```

File: src/main/java/core/basesyntax/Figure.java [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    }
+
+    public void setArea(double area) {
+        this.area = area;
+    }
+
+    @Override
+    public double getArea() {
+        return this.area;
+    }
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Figure.java:23]: This methods should override in the child
```suggestion
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 14]
```diff
@@ -0,0 +1,46 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier extends ColorSupplier {
+    private static final int MAX_VALUE_FOR_FIGURES = 26;
+    private static final int MAX_VALUE_FOR_CIRCLE_RADIUS = 10;
+    private Random random = new Random();
+
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/FigureSupplier.java:9]: This class "FigureSupplier" should not extends ColorSupplier
need to create a class variable
```suggestion
public class FigureSupplier {
    private static final int MAX_VALUE_FOR_FIGURES = 26;
    private static final int MAX_VALUE_FOR_CIRCLE_RADIUS = 10;
    private Random random = new Random();
    private ColorSupplier colorSupplier = new ColorSupplier();

```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    public Figure getRandomSquare() {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/FigureSupplier.java:10]: FigureSupplier should have only two public methods:
Знімок екрана 2023-01-22 о 13 33 18

Other methods that are used only in the class must be private

File: src/main/java/core/basesyntax/Main.java [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+                break;
+            case 5:
+                firstFigure.getRandomIsoscelesTrapezoid().draw();
+                break;
+            case 6:
+                firstFigure.getDefaultFigure().draw();
+                break;
+            default:
+                System.out.println("this number doesn't exist");
+        }
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Main.java:33]: This "swich" should be implemented in the FigureSupplier class in the getRandomFigure() method

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+public class FigureSupplier {
+    private static final int MAX_VALUE_FOR_FIGURES = 26;
+    private static final int MAX_VALUE_FOR_CIRCLE_RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    private Figure getRandomSquare() {
+        return new Square(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_VALUE_FOR_FIGURES));
+    }
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/FigureSupplier.java:15]: private methods should be below public ones

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+                firstFigure.getRandomSquare().draw();
+                break;
+            case 5:
+                firstFigure.getRandomIsoscelesTrapezoid().draw();
+                break;
+            case 6:
+                firstFigure.getDefaultFigure().draw();
+                break;
+            default:
+                System.out.println("this number doesn't exist");
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/FigureSupplier.java:74]: use the default method here
```suggestion
                return getDefaultFigure();
```

File: src/main/java/core/basesyntax/FigureSupplier.java [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+        return new Circle(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_VALUE_FOR_FIGURES));
+
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE, MAX_VALUE_FOR_CIRCLE_RADIUS);
+    }
+
+    public void getRandomFigure() {
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/FigureSupplier.java:49]: This method should return a Figure
```suggestion
    public Figure getRandomFigure() {
```

File: src/main/java/core/basesyntax/Main.java [hunk 20]
```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier firstFigure = new FigureSupplier();
+        FigureSupplier secondFigure = new FigureSupplier();
+        FigureSupplier thirdFigure = new FigureSupplier();
+        firstFigure.getRandomFigure();
+        secondFigure.getRandomFigure();
+        thirdFigure.getRandomFigure();
```
Comment (Ivan95kos) [inline src/main/java/core/basesyntax/Main.java:10]: use a loop to create random Figure, for this you only need one variable of type FigureSupplier

**Review by Ivan95kos — CHANGES_REQUESTED** [review]

**Review by okuzan — APPROVED** [review]
Good work!

**Pull request closed without merging**